src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/PackageSourceMappingUtility.cs

```diff
@@ -12,53 +12,80 @@ namespace NuGet.PackageManagement.UI
 {
     internal static class PackageSourceMappingUtility
     {
+        /// <summary>
+        /// Determines if new package source mappings should be written to the <paramref name="sourceMappingProvider"/> which is returned as a count.
+        /// </summary>
+        /// <param name="userAction"></param>
+        /// <param name="addedPackageIds"></param>
+        /// <param name="sourceMappingProvider"></param>
+        /// <param name="existingPackageSourceMappingSourceItems"></param>
+        /// <param name="countCreatedTopLevelSourceMappings">For Top-Level packages: <c>null</c> if not applicable; 0 if none needed to be added; > 0 is the count of new package source mappings added.</param>
+        /// <param name="countCreatedTransitiveSourceMappings">For Transitive packages: <c>null</c> if not applicable; 0 if none needed to be added; > 0 is the count of new package source mappings added.</param>
         internal static void ConfigureNewPackageSourceMapping(
             UserAction? userAction,
             IReadOnlyList<string>? addedPackageIds,
             PackageSourceMappingProvider sourceMappingProvider,
-            IReadOnlyList<PackageSourceMappingSourceItem> existingPackageSourceMappingSourceItems)
+            IReadOnlyList<PackageSourceMappingSourceItem> existingPackageSourceMappingSourceItems,
+            out int? countCreatedTopLevelSourceMappings,
```

Comment:
The 2 out parameters instinctively made me try to think of a better method separation. 

Would calling CreateAndSavePackageSourceMappings in a different method allow you to maybe just return the new source mapping items before saving them?
```cs
CreateAndSavePackageSourceMappings(
                userAction.SelectedSourceName,
                addedPackageIdsWithoutExistingMappings,
                sourceMappingProvider,
                existingPackageSourceMappingSourceItems);
```

Reply:
That makes sense to me, but I suggest addressing this comment in a subsequent PR so we can merge this PR in the next couple of days.

Reply:
Chatted with Nikolche offline: There's no simple refactoring that's meaningful right now at improving readability without significant changes. The goal of the utility from earlier feedback was to not continue growing the logic of within the UIActionEngine.